### PR TITLE
Get patient by demographic data poc

### PR DESCRIPTION
**Pull request by MatthewErispe**
## Type of PR (check all applicable)

- [ ] 💼 New Package
- [ ] ♻️ Refactor
- [x] ✨ New Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🧪 Tests
- [ ] 🔖 Release
- [ ] 🚩 Other

## Description

This PR adds patient demographics search and FHIR compliancy (search query + HTTP errors)

## Added to documentation?

- [x] 📜 readme
- [ ] 📓 general documentation
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Review by BMartinos — APPROVED** [review]
LGTM :+1:

File: projects/omang-service-mediator/src/modules/bdrs/repositories/birth.repository.ts [hunk 1]
```diff
@@ -310,12 +310,71 @@ export class BirthRepository {
...
+  }
 
-    return result;
+  async getByDemographicData(
+    firstName: string | null,
+    lastName: string | null,
+    gender: string | null,
+    birthDate: string | null,
+    pager: Pager,
+  ): Promise<BirthRecord[]> {
```
[the remark this answers is not in the document]
Comment (marrouchi) [reply projects/omang-service-mediator/src/modules/bdrs/repositories/birth.repository.ts:326]: Not if the patient is still alive 🕺

**Comment by marrouchi** [comment]
@MohamedAliBouhaouala All your comments have been addressed in the 2nd PR

File: projects/omang-service-mediator/src/modules/patient/controllers/patient.controller.ts [hunk 2]
```diff
@@ -37,48 +38,86 @@ export class PatientController {
...
   @Get('Get')
   @Header('Content-Type', 'application/fhir+json')
-  async get(@Query('identifier') identifier: string,
-  @Headers('x-openhim-clientid') clientId = 'OmangSvc'): Promise<fhirR4.Bundle> {
-    try {
-      if (!identifier || !identifier.includes('|')) {
-        throw new BadRequestException();
+  async get(
+    @Query('identifier') identifier: string,
+    @Query('given') givenNames: string, // Aligned with FHIR search parameters
```
[the remark this answers is not in the document]
Comment (marrouchi) [reply projects/omang-service-mediator/src/modules/patient/controllers/patient.controller.ts:43]: https://www.hl7.org/fhir/patient-search.html#8.1.37 

The spec says "The parameter cannot repeat or have multiple values"

**Pull request merged**